### PR TITLE
Clean up `passes/cmds/scatter.cc`.

### PR DESCRIPTION
**Pull request by boqwxp**

### CODE REVIEW

File: passes/cmds/scatter.cc [hunk 1]
```diff
@@ -46,22 +46,19 @@ struct ScatterPass : public Pass {
...
-		for (auto &mod_it : design->modules_)
+		for (auto module : design->selected_modules())
 		{
-			if (!design->selected(mod_it.second))
-				continue;
-
-			for (auto &c : mod_it.second->cells_)
-			for (auto &p : c.second->connections_)
+			for (auto cell : module->cells())
+			for (auto &p : cell->connections_)
```
Comment (nakengelhardt) [inline passes/cmds/scatter.cc:52]: This could also be `connections()`

Comment (boqwxp) [reply passes/cmds/scatter.cc]: Indeed, although I'm not sure b6eaff69dc28fa2dffd4b0572ca73b0ef823e143 counts as much of a cleanup :-\.

File: passes/cmds/scatter.cc [hunk 2]
```diff
@@ -46,25 +46,19 @@ struct ScatterPass : public Pass {
...
-				if (ct.cell_output(c.second->type, p.first)) {
-					RTLIL::SigSig sigsig(p.second, wire);
-					mod_it.second->connect(sigsig);
-				} else {
-					RTLIL::SigSig sigsig(wire, p.second);
-					mod_it.second->connect(sigsig);
+			for (auto cell : module->cells()) {
+				std::map<RTLIL::IdString, std::pair<RTLIL::SigSpec, RTLIL::SigSpec>> new_connections;
+				for (auto conn : cell->connections())
+					new_connections.emplace(conn.first, std::make_pair(conn.second, module->addWire(NEW_ID, conn.second.size())));
```
Comment (eddiehung) [inline passes/cmds/scatter.cc:54]:
```suggestion
			for (auto cell : module->cells()) {
				dict<RTLIL::IdString, RTLIL::SigSig> new_connections;
				for (auto conn : cell->connections())
					new_connections.insert(std::make_pair(conn.first, RTLIL::SigSig(conn.second, module->addWire(NEW_ID, conn.second.size()))));
```

**Comment by boqwxp** [comment]
Sorry for the commit spam. @eddiehung WDYT now?

**Comment by whitequark** [comment]
@boqwxp Should the `kernel/hashlib.h` changes be part of this PR?

**Comment by boqwxp** [comment]
Well, those were already merged, so... yes? :)

I just rebased on master, anyway.

**Comment by whitequark** [comment]
> Well, those were already merged, so... yes? :)

Mainly, I asked because having unrelated changes *and* several commits introducing related changes makes it hard to review the PR, since there is no easy way to see only the relevant ones.

**Comment by boqwxp** [comment]
Ah I see.  Unfortunately GitHub appears to have very poor UI for this.  I did actually submit a separate PR for the initial couple commits, and this PR depends on those changes, so I'm not sure how else it should be done.  But the only way you can see (AFAIK) that this is based on another PR is that some of the commits have a yellow ball/green checkmark/red X next to them, indicating they've been submitted to CI. :-\

Anyway if you manually checkout the branch and do `git log` you will see much more clearly that there are two different branch HEADs in the PR.

**Pull request merged**